proxy/http/HttpBodyFactory.cc

```diff
@@ -723,66 +687,42 @@ HttpBodyFactory::is_response_suppressed(HttpTransact::State *context)
 void
 HttpBodyFactory::nuke_template_tables()
 {
-  RawHashTable *h1, *h2;
-  RawHashTable_Value v1, v2;
-  RawHashTable_Binding *b1, *b2;
-  RawHashTable_IteratorState i1, i2;
-  HttpBodySet *body_set;
-  HttpBodyTemplate *hbt;
-
-  h1 = table_of_sets;
-
-  if (h1) {
+  if (table_of_sets) {
     Debug("body_factory", "deleting pre-existing template tables");
   } else {
     Debug("body_factory", "no pre-existing template tables");
   }
-  if (h1 != nullptr) {
+  if (table_of_sets) {
     ///////////////////////////////////////////
     // loop over set->body-types hash table //
     ///////////////////////////////////////////
-
-    for (b1 = h1->firstBinding(&i1); b1 != nullptr; b1 = h1->nextBinding(&i1)) {
-      v1 = h1->getValueFromBinding(b1);
-
-      body_set = (HttpBodySet *)v1;
+    for (auto it = table_of_sets->begin(); it != table_of_sets->end(); ++it) {
+      HttpBodySet *body_set = (HttpBodySet *)it->second;
       ink_assert(body_set->is_sane());
-      h2 = body_set->table_of_pages;
-
-      if (h2 != nullptr) {
-        body_set->table_of_pages = nullptr;
-
+      if (body_set->table_of_pages) {
         ///////////////////////////////////////////
         // loop over body-types->body hash table //
         ///////////////////////////////////////////
-
-        for (b2 = h2->firstBinding(&i2); b2 != nullptr; b2 = h2->nextBinding(&i2)) {
-          v2 = h2->getValueFromBinding(b2);
-          if (v2) {
-            // need a cast here
-            hbt = (HttpBodyTemplate *)v2;
-            delete hbt;
+        for (auto it_page = body_set->table_of_pages->begin(); it_page != body_set->table_of_pages->end(); ++it_page) {
+          if (it_page->second) {
```

Comment:
Just `delete it_page->second;` - it is legal and safe to call `delete` on `nullptr`.